### ISSUE TITLE
PlayableAssetState causes exception in Unity 2021.1

### ISSUE THREAD

**Issue by KybernetikGames** [KybernetikGames/animancer#111]
# Environment

- Animancer Version Number: 6.1
- Animancer Pro
- Unity Version: 2021.1
- Timeline Package: 1.5.4

# Description

Using a `PlayableAssetState` causes an exception every frame in Unity 2021.1.

Unity 2020.3 and older don't have the same issue.

# Reproduction

Steps to reproduce the bug:

1. Create new empty project in Unity 2021.1.
2. Download Timeline 1.5.4 and Animancer v6.1.
3. Run example scene from Plugins\Animancer\Examples\06 State Machines\07 Platformer
4. The following error is logged every frame:

ArgumentOutOfRangeException: Specified argument was out of the range of valid values.
Parameter name: Time is over the discrete range.
UnityEngine.Timeline.DiscreteTime.DoubleToDiscreteTime (System.Double time) (at Library/PackageCache/com.unity.timeline@1.5.4/Runtime/DiscreteTime.cs:88)
UnityEngine.Timeline.DiscreteTime..ctor (System.Double time) (at Library/PackageCache/com.unity.timeline@1.5.4/Runtime/DiscreteTime.cs:26)
UnityEngine.Timeline.TimelinePlayable.Evaluate (UnityEngine.Playables.Playable playable, UnityEngine.Playables.FrameData frameData) (at Library/PackageCache/com.unity.timeline@1.5.4/Runtime/TimelinePlayable.cs:265)
UnityEngine.Timeline.TimelinePlayable.PrepareFrame (UnityEngine.Playables.Playable playable, UnityEngine.Playables.FrameData info) (at Library/PackageCache/com.unity.timeline@1.5.4/Runtime/TimelinePlayable.cs:244)

**Comment by KybernetikGames** [comment]
1. Open Assets/Plugins/Animancer/Internal/Core/PlayableAssetState.cs.
2. Go to the CreatePlayable method.
3. Replace the line _Length = (float)_Asset.duration; with the following:

```
var duration = _Asset.duration;
playable.SetDuration(duration);
_Length = (float)duration;
```

Yeah ... the bug was just that the Timeline doesn't assign its duration to the Playable it creates.

**Comment by KybernetikGames** [comment]
Apparently setting the Playable duration affects the way it tracks time. Normally the time keeps advancing after a state ends and Animancer Events rely on that behaviour, but if you set the duration it stops at that point. In particular, an end event at normalized time 1 will never get triggered because the normalized time never passes 1.

So instead, I went and figured out the maximum allowed value which is:

```
playable.SetDuration(9223372.03685477);
_Length = (float)_Asset.duration;
```

**Comment by KybernetikGames** [comment]
This fix is included in Animancer v6.1 which is now available on [itch.io](https://kybernetik.itch.io/animancer-pro) and the [Asset Store](https://assetstore.unity.com/packages/tools/animation/animancer-pro-116514?aid=1100l8ah5&utm_source=aff).
